### PR TITLE
Fixed various minor grammatical issues

### PR DESCRIPTION
**Pull request by PtrStruct**
Various issues were fixed as well as cleared up for the reader to understand the document a little bit better.

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/csharp/programming-guide/classes-and-structs/classes.md [hunk 1]
```diff
@@ -11,19 +11,23 @@ ms.assetid: e8848524-7273-429f-8aba-c658d5eff5ad
...
+When the object is created, enough memory is allocated on the managed heap for that specific object, and the variable holds only a reference to the location of said object. Types on the managed heap require overhead both when they are allocated and when they are reclaimed by the automatic memory management functionality of the CLR, which is known as *garbage collection*. However, garbage collection is also highly optimized and in most scenarios, it does not create a performance issue. For more information about garbage collection, see [Automatic memory management and garbage collection](../../../standard/garbage-collection/gc.md).  
   
 ## Declaring Classes  
- Classes are declared by using the [class](../../../csharp/language-reference/keywords/class.md) keyword, as shown in the following example:
+ Classes are declared by using the [class](../../../csharp/language-reference/keywords/class.md) keyword followed by a unique identifier, as shown in the following example:
 
  ```csharp
+[access modifier] - [class] - [identifier]
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/classes.md]: I like this, but I'd like this line to be a comment so it's not interpreted as an Attribute:

```csharp
// [access modifier] - [class] - [identifier]
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making these updates @PtrStruct 
Overall, this is a great set of changes.

I did have a couple suggestions before we merge this.

Thanks again.

File: docs/csharp/programming-guide/classes-and-structs/classes.md (hunk 1 above)
Comment (PtrStruct) [reply docs/csharp/programming-guide/classes-and-structs/classes.md]: Perfect! I was pretty sure I added it as a comment but I guess it slipped my eye!

**Comment by PtrStruct** [comment]
@BillWagner Is there anything else you would like me to change or does it all look good to you?

File: docs/csharp/programming-guide/classes-and-structs/classes.md [hunk 2]
```diff
@@ -11,19 +11,23 @@ ms.assetid: e8848524-7273-429f-8aba-c658d5eff5ad
...
 
 ## Reference types  
-A type that is defined as a [class](../../../csharp/language-reference/keywords/class.md) is a *reference type*. At run time, when you declare a variable of a reference type, the variable contains the value [null](../../../csharp/language-reference/keywords/null.md) until you explicitly create an instance of the class by using the [new](../../../csharp/language-reference/keywords/new.md) operator, or assign it an object that has been created elsewhere, as shown in the following example:
+A type that is defined as a [class](../../../csharp/language-reference/keywords/class.md) is a *reference type*. At run time, when you declare a variable of a reference type, the variable contains the value [null](../../../csharp/language-reference/keywords/null.md) until you explicitly create an instance of the class by using the [new](../../../csharp/language-reference/keywords/new.md) operator, or assign it an object of the same type that may have been created elsewhere, as shown in the following example:
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/classes.md]: Here, where you say "an object of the same type", that should say "an object of a compatible type". These assignments are all valid:

```csharp
string s = "hello";
object o = s; // related by inheritance
List<int> collection = new List<int>();
IEnumerable<int> enumerable = collection; // related by interface implementation.
```

**Comment by BillWagner** [comment]
@PtrStruct 

> Is there anything else you would like me to change or does it all look good to you?

I added one comment on L 14. (I looks like it got lost when I submitted the earlier review.) Once that is updated, I'll :shipit:

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @PtrStruct 

I'll :shipit: as soon as the build completes.

File: docs/csharp/programming-guide/classes-and-structs/classes.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ ms.assetid: e8848524-7273-429f-8aba-c658d5eff5ad
...
 
 ## Reference types  
-A type that is defined as a [class](../../../csharp/language-reference/keywords/class.md) is a *reference type*. At run time, when you declare a variable of a reference type, the variable contains the value [null](../../../csharp/language-reference/keywords/null.md) until you explicitly create an instance of the class by using the [new](../../../csharp/language-reference/keywords/new.md) operator, or assign it an object of the same type that may have been created elsewhere, as shown in the following example:
+A type that is defined as a [class](../../../csharp/language-reference/keywords/class.md) is a *reference type*. At run time, when you declare a variable of a reference type, the variable contains the value [null](../../../csharp/language-reference/keywords/null.md) until you explicitly create an instance of the class by using the [new](../../../csharp/language-reference/keywords/new.md) operator, or assign it an object of an object of a compatible type that may have been created elsewhere, as shown in the following example:
```
Comment (pkulikov) [inline docs/csharp/programming-guide/classes-and-structs/classes.md]: typo: "object of an object of a compatible type"

**Pull request merged**